### PR TITLE
Add farmer npc

### PR DESCRIPTION
**Pull request by OgelGames**
Adds farmers that harvest and plant crops just as a player would.

screenshot_20210111_005335

**Features/abilities**
* Harvests and replants any ripe crop
* Plants new crops on dirt if it has seeds and water is near
* Can be given seeds to plant
* Gives harvested items to the player when right-clicked with a gold lump
* Can be "locked" when right-clicked with paper (stops other players from taking items from farmers)
* Can be tamed, captured, protected, named, and ordered like other NPCs
* Drops most of harvested items when killed

**Testing:**
* [x] Can plant every different crop
* [x] Can harvest every different crop
* [x] Does not duplicate items (i.e. not a mithril scythe)
* [x] Does not destroy items (except when killed)
* [x] Can be tamed, captured, protected, named, and ordered

@BuckarooBanzay @SwissalpS @6r1d @S-S-X

### CODE REVIEW

File: farmer.lua [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+	["farming:rhubarb_3"] = {"farming:rhubarb_1", "farming:rhubarb"},
+	["farming:rice_8"] = {"farming:rice_1", "farming:seed_rice"},
+	["farming:rye_8"] = {"farming:rye_1", "farming:seed_rye"},
+	["farming:soy_7"] = {"farming:soy_1", "farming:soy_pod"},
+	["farming:tomato_8"] = {"farming:tomato_1", "farming:tomato"},
+	["farming:wheat_8"] = {"farming:wheat_1", "farming:seed_wheat"},
+	["farming:vanilla_8"] = {"farming:vanilla_1", "farming:vanilla"},
+}
+
+-- Itsms that need to be crafted into seeds before planting
```
Comment (SwissalpS) [inline farmer.lua]: typo: Items not Itsms

**Comment by SwissalpS** [comment]
I don't see handler for cocoa, which does not need soil. You are probably aware of this already.

**Comment by OgelGames** [comment]
> I don't see handler for cocoa, which does not need soil. You are probably aware of this already.

Hmm... Well originally I found it was too different to include, but it would be nice to find a way to work it in, just for completeness... :thinking:

**Comment by SwissalpS** [comment]
It's the one crop I would first want to have automation for :)
If you want to add others for completeness: agave, mushrooms and flowers would be interesting too. Also cactus and papyrus.
Most of these need individual treatment.

**Comment by OgelGames** [comment]
> If you want to add others for completeness: agave, mushrooms and flowers would be interesting too. Also cactus and papyrus.

Those are not part of `farming`, and also grow differently, so I'm not counting those 😉

**Comment by OgelGames** [comment]
I think I'll merge this soon, I did some more testing, and there were no crashes or bugs, so I think this should be safe for "live testing" 😉

**Comment by SwissalpS** [comment]
I have added to tutorial and have spawning grounds ready :)

**Comment by BuckarooBanzay** [comment]
I've played with them a little, the idea is nice and they do as advertised, just the handling/interface is a bit minimalistic :yum:

**Review by BuckarooBanzay — APPROVED** [review]

**Pull request merged**